compair/tests/algorithms/test_pair_adaptive_min_delta.py

```diff
@@ -0,0 +1,462 @@
+        self.assertIn(min_key, [1, 2, 3])
+        self.assertEqual(max_key, 4)
+
+        ##
+        # Can select previously compared object but not with same opponent
+        scored_objects = [
+            ScoredObject(
+                key=1, score=0.5, variable1=None, variable2=None,
+                rounds=2, wins=None, loses=None, opponents=None
+            ),
+            ScoredObject(
+                key=2, score=0.5, variable1=None, variable2=None,
+                rounds=2, wins=None, loses=None, opponents=None
+            ),
+            ScoredObject(
+                key=3, score=0.5, variable1=None, variable2=None,
+                rounds=3, wins=None, loses=None, opponents=None
+            )
+        ]
+        comparisons = [
+            ComparisonPair(1,2,None)
+        ]
+        criterion_scores = {
+            1: { 'c1': 10.1, 'c2': 20.2 },
+            2: { 'c1': 10.1, 'c2': 20.2 },
+            3: { 'c1': 10.1, 'c2': 20.2 }
+        }
+        criterion_weights = {
+            'c1': 0.4, 'c2': 0.6
+        }
+
+        results = self.pair_algorithm.generate_pair(scored_objects, comparisons,
+            criterion_scores, criterion_weights)
+
+        min_key = min([results.key1, results.key2])
+        max_key = max([results.key1, results.key2])
+        self.assertEqual(min_key, 1)
+        self.assertEqual(max_key, 3)
+
+        ##
+        # Select opponent with closest score (mock shuffle order)
+        scored_objects = [
+            ScoredObject(
+                key=1, score=0.5, variable1=None, variable2=None,
+                rounds=3, wins=None, loses=None, opponents=None
+            ),
+            ScoredObject(
+                key=2, score=0.7, variable1=None, variable2=None,
+                rounds=3, wins=None, loses=None, opponents=None
+            ),
+            ScoredObject(
+                key=3, score=0.2, variable1=None, variable2=None,
+                rounds=3, wins=None, loses=None, opponents=None
+            ),
+            ScoredObject(
+                key=4, score=0.4, variable1=None, variable2=None,
+                rounds=3, wins=None, loses=None, opponents=None
+            )
+        ]
+        comparisons = []
+        criterion_scores = {
+            1: { 'c1': 10.1, 'c2': 20.2 },
+            2: { 'c1': 10.1, 'c2': 20.2 },
+            3: { 'c1': 10.1, 'c2': 20.2 },
+            4: { 'c1': 10.1, 'c2': 20.2 }
+        }
+        criterion_weights = {
+            'c1': 0.4, 'c2': 0.6
+        }
+        results = self.pair_algorithm.generate_pair(scored_objects, comparisons,
+            criterion_scores, criterion_weights)
+
+        # object 1 should be selected since random shuffle is disabled
+        self.assertEqual(results.key1, 1)
+        # object 4 should be selected as 0.4 is the closest value to 0.5
+        self.assertEqual(results.key2, 4)
+
+        ##
+        # Select opponent with closest score (mock shuffle order)
```

Comment:
add criteria score test in same spirit

Reply:
test case added
https://github.com/ubc/compair/blob/3837f3209f1394e878c852180b7667b9355cb69b/compair/tests/algorithms/test_pair_adaptive_min_delta.py#L468-L508